src/Record.ts

```diff
@@ -38,6 +39,10 @@ export class ReadableRecord {
     this.labels = labels;
     this.contentType = contentType;
     this.arrayBuffer = arrayBuffer;
+
+    if (head) {
+      (this.arrayBuffer as ArrayBuffer) = new ArrayBuffer(0);
```

Comment:
Nice okay, so when head is true it will return an empty buffer.
Would it also work with "this.stream.push(null);" to end the stream? 
I didn't fully get why the head request is hanging, but wondering if this also fixes it for batch requests.